doc_gen.php

```diff
@@ -0,0 +1,164 @@
+<?php
```

Comment:
i think we can move this to `doc/generate.php`, and add it to [.php_cs.dist](https://github.com/azjezz/psl/blob/1.5.x/.php_cs.dist#L7) and [.phpcs.xml](https://github.com/azjezz/psl/blob/1.5.x/.phpcs.xml#L6) to ensure it follows our CS rules.

Reply:
Currently all tasks are also in `composer.json`. Maybe add a shortcut in there as well and add it to the `composer ci` task?